### PR TITLE
chore(issue-priority): Update documentation for enhanced issue priority

### PR DESCRIPTION
**Pull request by snigdhas**
Updating issue priority documentation to explain the feature differences between free/team plans and business plans. 

Fixes https://github.com/getsentry/sentry/issues/74979

### CODE REVIEW

File: docs/product/issues/issue-priority/index.mdx [hunk 1]
```diff
@@ -8,14 +8,24 @@ Issue priority sorts the issues Sentry receives into **High**, **Medium** and **
...
-When Sentry receives an event, it assigns a priority to the issue based on the event's **log level** (for error issues) or **actionability** (for non-error issues like performance). For Python and Javascript projects, Sentry will also consider the error message, whether or not the error is handled, and historical actions taken on similar issues.
+When Sentry receives an event, it assigns a priority to the issue based on the event's **log level** (for error issues) or **actionability** (for non-error issues like performance).
 
 There are three discrete priority levels for issues in Sentry:
 
 - **High**: Issues that are likely to be actionable and require immediate attention.
 - **Medium**: Issues that are likely to be actionable and require attention in the near future.
 - **Low**: Issues that don't require immediate attention.
 
+### Enhanced Issue Priority and Alerting
```
Comment (mrduncan) [inline docs/product/issues/issue-priority/index.mdx:19]:
```suggestion
### Enhanced Issue Priority
```
nit: we don't talk about alerting in this section

File: docs/product/issues/issue-priority/index.mdx [hunk 2]
```diff
@@ -8,14 +8,24 @@ Issue priority sorts the issues Sentry receives into **High**, **Medium** and **
...
+
+<Note>
+
+This feature is available only if your organization is on either a Business or Trial Plan.
+
+</Note>
+
+The enhanced priority calculations will consider additional factors when determining the priority of an issue. In addition to the log level or actionability, Sentry will also consider the error message, whether or not the error is handled, and historical actions taken on similar issues.
```
Comment (mrduncan) [inline docs/product/issues/issue-priority/index.mdx:27]: Should we specify this is only the case for python/javascript?

File: docs/product/issues/issue-priority/index.mdx [hunk 3]
```diff
@@ -8,14 +8,24 @@ Issue priority sorts the issues Sentry receives into **High**, **Medium** and **
...
 
 - **High**: Issues that are likely to be actionable and require immediate attention.
 - **Medium**: Issues that are likely to be actionable and require attention in the near future.
 - **Low**: Issues that don't require immediate attention.
 
+### Enhanced Issue Priority
+
+<Note>
+
+This feature is available only if your organization is on either a Business or Trial Plan.
```
Comment (lizokm) [inline docs/product/issues/issue-priority/index.mdx:23]:
```suggestion
This feature is available only if your organization is on either a [Business](https://sentry.io/pricing/) or [Trial](/pricing/#product-trials) Plan.
```

File: docs/product/issues/issue-priority/index.mdx [hunk 4]
```diff
@@ -8,14 +8,24 @@ Issue priority sorts the issues Sentry receives into **High**, **Medium** and **
...
+
+</Note>
+
+The enhanced priority calculations will consider additional factors when determining the priority of an issue for Python or JavaScript projects. In addition to the log level or actionability, Sentry will also consider the error message, whether or not the error is handled, and historical actions taken on similar issues.
+
```
Comment (lizokm) [inline docs/product/issues/issue-priority/index.mdx:28]:
```suggestion
In addition to the log level or actionability, the enhanced priority calculations will consider factors such as the error message, whether or not the error is handled, and historical actions taken on similar issues, when determining the priority of an issue for Python or JavaScript projects.

```

**Review by lizokm — APPROVED** [review]
Thanks for updating!!

**Pull request merged**